File: Models/ApsimCampbell/src/java/ApsimCampbell/SoiltempExogenous.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class SoiltempExogenous
{
    private double waterBalance_Eo;
    private double waterBalance_Salb;
    private Double [] organic_Carbon;
    private double waterBalance_Es;
    private double weather_Wind;
    private Double [] physical_ParticleSizeSand;
    private double weather_AirPressure;
    private Integer clock_Today_DayOfYear;
    private double microClimate_CanopyHeight;
    private double waterBalance_Eos;
    private Double [] waterBalance_SW;
    private double weather_Amp;
    private double weather_MinT;
    private double weather_Radn;
    private Double [] physical_Rocks;
    private double weather_Tav;
    private double weather_MaxT;
    private double weather_MeanT;
    private Double [] physical_ParticleSizeSilt;
    private Double [] physical_ParticleSizeClay;
    
    public SoiltempExogenous() { }
    
    public SoiltempExogenous(SoiltempExogenous toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.waterBalance_Eo = toCopy.getwaterBalance_Eo();
            this.waterBalance_Salb = toCopy.getwaterBalance_Salb();
            organic_Carbon = new Double[toCopy.getorganic_Carbon().length];
        for (int i = 0; i < toCopy.getorganic_Carbon().length; i++)
        {
            organic_Carbon[i] = toCopy.getorganic_Carbon()[i];
        }
            this.waterBalance_Es = toCopy.getwaterBalance_Es();
            this.weather_Wind = toCopy.getweather_Wind();
            physical_ParticleSizeSand = new Double[toCopy.getphysical_ParticleSizeSand().length];
        for (int i = 0; i < toCopy.getphysical_ParticleSizeSand().length; i++)
        {
            physical_ParticleSizeSand[i] = toCopy.getphysical_ParticleSizeSand()[i];
        }
            this.weather_AirPressure = toCopy.getweather_AirPressure();
            this.clock_Today_DayOfYear = toCopy.getclock_Today_DayOfYear();
            this.microClimate_CanopyHeight = toCopy.getmicroClimate_CanopyHeight();
            this.waterBalance_Eos = toCopy.getwaterBalance_Eos();
            waterBalance_SW = new Double[toCopy.getwaterBalance_SW().length];
        for (int i = 0; i < toCopy.getwaterBalance_SW().length; i++)
        {
            waterBalance_SW[i] = toCopy.getwaterBalance_SW()[i];
        }
            this.weather_Amp = toCopy.getweather_Amp();
            this.weather_MinT = toCopy.getweather_MinT();
            this.weather_Radn = toCopy.getweather_Radn();
            physical_Rocks = new Double[toCopy.getphysical_Rocks().length];
        for (int i = 0; i < toCopy.getphysical_Rocks().length; i++)
        {
            physical_Rocks[i] = toCopy.getphysical_Rocks()[i];
        }
            this.weather_Tav = toCopy.getweather_Tav();
            this.weather_MaxT = toCopy.getweather_MaxT();
            this.weather_MeanT = toCopy.getweather_MeanT();
            physical_ParticleSizeSilt = new Double[toCopy.getphysical_ParticleSizeSilt().length];
        for (int i = 0; i < toCopy.getphysical_ParticleSizeSilt().length; i++)
        {
            physical_ParticleSizeSilt[i] = toCopy.getphysical_ParticleSizeSilt()[i];
        }
            physical_ParticleSizeClay = new Double[toCopy.getphysical_ParticleSizeClay().length];
        for (int i = 0; i < toCopy.getphysical_ParticleSizeClay().length; i++)
        {
            physical_ParticleSizeClay[i] = toCopy.getphysical_ParticleSizeClay()[i];
        }
        }
    }
    public double getwaterBalance_Eo()
    { return waterBalance_Eo; }

    public void setwaterBalance_Eo(double _waterBalance_Eo)
    { this.waterBalance_Eo= _waterBalance_Eo; } 
    
    public double getwaterBalance_Salb()
    { return waterBalance_Salb; }

    public void setwaterBalance_Salb(double _waterBalance_Salb)
    { this.waterBalance_Salb= _waterBalance_Salb; } 
    
    public Double [] getorganic_Carbon()
    { return organic_Carbon; }

    public void setorganic_Carbon(Double [] _organic_Carbon)
    { this.organic_Carbon= _organic_Carbon; } 
    
    public double getwaterBalance_Es()
    { return waterBalance_Es; }

    public void setwaterBalance_Es(double _waterBalance_Es)
    { this.waterBalance_Es= _waterBalance_Es; } 
    
    public double getweather_Wind()
    { return weather_Wind; }

    public void setweather_Wind(double _weather_Wind)
    { this.weather_Wind= _weather_Wind; } 
    
    public Double [] getphysical_ParticleSizeSand()
    { return physical_ParticleSizeSand; }

    public void setphysical_ParticleSizeSand(Double [] _physical_ParticleSizeSand)
    { this.physical_ParticleSizeSand= _physical_ParticleSizeSand; } 
    
    public double getweather_AirPressure()
    { return weather_AirPressure; }

    public void setweather_AirPressure(double _weather_AirPressure)
    { this.weather_AirPressure= _weather_AirPressure; } 
    
    public Integer getclock_Today_DayOfYear()
    { return clock_Today_DayOfYear; }

    public void setclock_Today_DayOfYear(Integer _clock_Today_DayOfYear)
    { this.clock_Today_DayOfYear= _clock_Today_DayOfYear; } 
    
    public double getmicroClimate_CanopyHeight()
    { return microClimate_CanopyHeight; }

    public void setmicroClimate_CanopyHeight(double _microClimate_CanopyHeight)
    { this.microClimate_CanopyHeight= _microClimate_CanopyHeight; } 
    
    public double getwaterBalance_Eos()
    { return waterBalance_Eos; }

    public void setwaterBalance_Eos(double _waterBalance_Eos)
    { this.waterBalance_Eos= _waterBalance_Eos; } 
    
    public Double [] getwaterBalance_SW()
    { return waterBalance_SW; }

    public void setwaterBalance_SW(Double [] _waterBalance_SW)
    { this.waterBalance_SW= _waterBalance_SW; } 
    
    public double getweather_Amp()
    { return weather_Amp; }

    public void setweather_Amp(double _weather_Amp)
    { this.weather_Amp= _weather_Amp; } 
    
    public double getweather_MinT()
    { return weather_MinT; }

    public void setweather_MinT(double _weather_MinT)
    { this.weather_MinT= _weather_MinT; } 
    
    public double getweather_Radn()
    { return weather_Radn; }

    public void setweather_Radn(double _weather_Radn)
    { this.weather_Radn= _weather_Radn; } 
    
    public Double [] getphysical_Rocks()
    { return physical_Rocks; }

    public void setphysical_Rocks(Double [] _physical_Rocks)
    { this.physical_Rocks= _physical_Rocks; } 
    
    public double getweather_Tav()
    { return weather_Tav; }

    public void setweather_Tav(double _weather_Tav)
    { this.weather_Tav= _weather_Tav; } 
    
    public double getweather_MaxT()
    { return weather_MaxT; }

    public void setweather_MaxT(double _weather_MaxT)
    { this.weather_MaxT= _weather_MaxT; } 
    
    public double getweather_MeanT()
    { return weather_MeanT; }

    public void setweather_MeanT(double _weather_MeanT)
    { this.weather_MeanT= _weather_MeanT; } 
    
    public Double [] getphysical_ParticleSizeSilt()
    { return physical_ParticleSizeSilt; }

    public void setphysical_ParticleSizeSilt(Double [] _physical_ParticleSizeSilt)
    { this.physical_ParticleSizeSilt= _physical_ParticleSizeSilt; } 
    
    public Double [] getphysical_ParticleSizeClay()
    { return physical_ParticleSizeClay; }

    public void setphysical_ParticleSizeClay(Double [] _physical_ParticleSizeClay)
    { this.physical_ParticleSizeClay= _physical_ParticleSizeClay; } 
    
}

File: Models/ApsimCampbell/src/cs/ApsimCampbell/SoiltempExogenous.cs
using System;
using System.Collections.Generic;

public class SoiltempExogenous 
{
    private double _weather_MaxT;
    private double[] _waterBalance_SW;
    private int _clock_Today_DayOfYear;
    private double _weather_Radn;
    private double _microClimate_CanopyHeight;
    private double _weather_Latitude;
    private double _weather_MeanT;
    private double[] _physical_Rocks;
    private double _waterBalance_Eo;
    private double _timestep;
    private double _weather_MinT;
    private double[] _organic_Carbon;
    private double _weather_Tav;
    private double[] _physical_ParticleSizeSand;
    private double _weather_AirPressure;
    private double _waterBalance_Eos;
    private double _waterBalance_Es;
    private double _waterBalance_Salb;
    private double[] _physical_ParticleSizeSilt;
    private double _weather_Wind;
    private double _weather_Amp;
    private double[] _physical_ParticleSizeClay;
    
    /// <summary>
    /// Constructor of the SoiltempExogenous component")
    /// </summary>  
    public SoiltempExogenous() { }
    
    
    public SoiltempExogenous(SoiltempExogenous toCopy, bool copyAll) // copy constructor 
    {
        if (copyAll)
        {
    
            weather_MaxT = toCopy.weather_MaxT;
            waterBalance_SW = new double[toCopy.waterBalance_SW.Length];
        for (int i = 0; i < toCopy.waterBalance_SW.Length; i++)
            { waterBalance_SW[i] = toCopy.waterBalance_SW[i]; }
    
            clock_Today_DayOfYear = toCopy.clock_Today_DayOfYear;
            weather_Radn = toCopy.weather_Radn;
            microClimate_CanopyHeight = toCopy.microClimate_CanopyHeight;
            weather_Latitude = toCopy.weather_Latitude;
            weather_MeanT = toCopy.weather_MeanT;
            physical_Rocks = new double[toCopy.physical_Rocks.Length];
        for (int i = 0; i < toCopy.physical_Rocks.Length; i++)
            { physical_Rocks[i] = toCopy.physical_Rocks[i]; }
    
            waterBalance_Eo = toCopy.waterBalance_Eo;
            timestep = toCopy.timestep;
            weather_MinT = toCopy.weather_MinT;
            organic_Carbon = new double[toCopy.organic_Carbon.Length];
        for (int i = 0; i < toCopy.organic_Carbon.Length; i++)
            { organic_Carbon[i] = toCopy.organic_Carbon[i]; }
    
            weather_Tav = toCopy.weather_Tav;
            physical_ParticleSizeSand = new double[toCopy.physical_ParticleSizeSand.Length];
        for (int i = 0; i < toCopy.physical_ParticleSizeSand.Length; i++)
            { physical_ParticleSizeSand[i] = toCopy.physical_ParticleSizeSand[i]; }
    
            weather_AirPressure = toCopy.weather_AirPressure;
            waterBalance_Eos = toCopy.waterBalance_Eos;
            waterBalance_Es = toCopy.waterBalance_Es;
            waterBalance_Salb = toCopy.waterBalance_Salb;
            physical_ParticleSizeSilt = new double[toCopy.physical_ParticleSizeSilt.Length];
        for (int i = 0; i < toCopy.physical_ParticleSizeSilt.Length; i++)
            { physical_ParticleSizeSilt[i] = toCopy.physical_ParticleSizeSilt[i]; }
    
            weather_Wind = toCopy.weather_Wind;
            weather_Amp = toCopy.weather_Amp;
            physical_ParticleSizeClay = new double[toCopy.physical_ParticleSizeClay.Length];
        for (int i = 0; i < toCopy.physical_ParticleSizeClay.Length; i++)
            { physical_ParticleSizeClay[i] = toCopy.physical_ParticleSizeClay[i]; }
    
        }
    }
    public double weather_MaxT
    {
        get { return this._weather_MaxT; }
        set { this._weather_MaxT= value; } 
    }
    public double[] waterBalance_SW
    {
        get { return this._waterBalance_SW; }
        set { this._waterBalance_SW= value; } 
    }
    public int clock_Today_DayOfYear
    {
        get { return this._clock_Today_DayOfYear; }
        set { this._clock_Today_DayOfYear= value; } 
    }
    public double weather_Radn
    {
        get { return this._weather_Radn; }
        set { this._weather_Radn= value; } 
    }
    public double microClimate_CanopyHeight
    {
        get { return this._microClimate_CanopyHeight; }
        set { this._microClimate_CanopyHeight= value; } 
    }
    public double weather_Latitude
    {
        get { return this._weather_Latitude; }
        set { this._weather_Latitude= value; } 
    }
    public double weather_MeanT
    {
        get { return this._weather_MeanT; }
        set { this._weather_MeanT= value; } 
    }
    public double[] physical_Rocks
    {
        get { return this._physical_Rocks; }
        set { this._physical_Rocks= value; } 
    }
    public double waterBalance_Eo
    {
        get { return this._waterBalance_Eo; }
        set { this._waterBalance_Eo= value; } 
    }
    public double timestep
    {
        get { return this._timestep; }
        set { this._timestep= value; } 
    }
    public double weather_MinT
    {
        get { return this._weather_MinT; }
        set { this._weather_MinT= value; } 
    }
    public double[] organic_Carbon
    {
        get { return this._organic_Carbon; }
        set { this._organic_Carbon= value; } 
    }
    public double weather_Tav
    {
        get { return this._weather_Tav; }
        set { this._weather_Tav= value; } 
    }
    public double[] physical_ParticleSizeSand
    {
        get { return this._physical_ParticleSizeSand; }
        set { this._physical_ParticleSizeSand= value; } 
    }
    public double weather_AirPressure
    {
        get { return this._weather_AirPressure; }
        set { this._weather_AirPressure= value; } 
    }
    public double waterBalance_Eos
    {
        get { return this._waterBalance_Eos; }
        set { this._waterBalance_Eos= value; } 
    }
    public double waterBalance_Es
    {
        get { return this._waterBalance_Es; }
        set { this._waterBalance_Es= value; } 
    }
    public double waterBalance_Salb
    {
        get { return this._waterBalance_Salb; }
        set { this._waterBalance_Salb= value; } 
    }
    public double[] physical_ParticleSizeSilt
    {
        get { return this._physical_ParticleSizeSilt; }
        set { this._physical_ParticleSizeSilt= value; } 
    }
    public double weather_Wind
    {
        get { return this._weather_Wind; }
        set { this._weather_Wind= value; } 
    }
    public double weather_Amp
    {
        get { return this._weather_Amp; }
        set { this._weather_Amp= value; } 
    }
    public double[] physical_ParticleSizeClay
    {
        get { return this._physical_ParticleSizeClay; }
        set { this._physical_ParticleSizeClay= value; } 
    }
}